department-of-veterans-affairs/vets-website
# Arf 84639 Form 21a Architecture

## Summary

- Summarize the changes that have been made to the platform: 
  -  In order to make the the 21a form application more maintainable the following file architecture was created:
    - The folders in the pages directory are the chapters (aka steps in the UI)
    - Each chapter folder begins with its step number
    - Each chapter has an index.js file that includes its schema with its title (populates the title in the step counter) and a object with keys of the pages
    - The pages are in their own files in the chapter folder
  - The folders, index.js files, and 1 page file for each step were created to self document this architecture through the code
  - I initially created a placeholder page that all chapters shared but to better demo functionality and to also provide value I made each page the first page of the step which conveniently only had description content. In order to align with [designs](https://www.figma.com/design/2afIGOMII0uRI5ck1dWo1w/Form-21a---Apply-for-Accreditation-(CA-%26-Attorneys)?node-id=123-22442&t=bwiMZL3TXTOhEQWU-4) the descriptions required creating a react component for 3 of them. 
- Why is this the solution? The form.js is very clean with just references to the chapters. The chapters are clean with just references to its pages. The pages contain the bulk of the logic.
- Which team do you work for, does your team own the maintenance of this component?
  - Accredited Representative Facing (#benefits-representative-facing in Slack)
  - Yes we will own Form 21a

## Related issue(s)

- [Form 21a Form Architecture, Introduction, and Confirmation #84639](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/84639)
- [[ARF] Digital 21a](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/83627)

## Testing done

- Navigate from Introduction page through all 7 steps to review and submit
- Ensure the pages align with [designs](https://www.figma.com/design/2afIGOMII0uRI5ck1dWo1w/Form-21a---Apply-for-Accreditation-(CA-%26-Attorneys)?node-id=123-22442&t=bwiMZL3TXTOhEQWU-4)

## Screenshots

### Before - Pages did not exist

### After
#### Mobile
`accreditation/21a/pages/02-employment-information-chapter/yourEmployers.js`
<img width="384" alt="Screenshot 2024-06-05 at 11 30 10 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/3d44f26d-c4c8-4c70-a8f1-3cda97a1999a">

`accreditation/21a/pages/03-education-history-chapter/yourEducationHistory.js`
<img width="389" alt="Screenshot 2024-06-05 at 11 30 25 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/a3f21f04-9126-4ec3-a294-f952ef7900b9">

`accreditation/21a/pages/04-law-practice-information-chapter/yourPracticingInformation.js`
<img width="386" alt="Screenshot 2024-06-05 at 11 30 39 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/9b8c201a-f5f3-4d7d-8948-d62a939c3573">

`accreditation/21a/pages/05-background-information-chapter/yourBackgroundInformation.js`
<img width="390" alt="Screenshot 2024-06-05 at 11 30 49 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/f370df21-779d-4cad-9ab5-ad5191930ea4">

`accreditation/21a/pages/06-character-references-chapter/yourCharacterReferences.js`
<img width="386" alt="Screenshot 2024-06-05 at 11 31 08 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/0d0cd763-6a1f-46bf-a6c6-216c5b517e1a">

#### Desktop
`accreditation/21a/pages/02-employment-information-chapter/yourEmployers.js`
<img width="612" alt="Screenshot 2024-06-05 at 11 27 45 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/e7bbf558-bfb8-4d70-8942-70ed668f5577">

`accreditation/21a/pages/03-education-history-chapter/yourEducationHistory.js`
<img width="599" alt="Screenshot 2024-06-05 at 11 28 44 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/d9757d67-cc14-4867-9780-2e0407aadab0">

`accreditation/21a/pages/04-law-practice-information-chapter/yourPracticingInformation.js`
<img width="611" alt="Screenshot 2024-06-05 at 11 28 59 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/60f3b521-b83e-4567-bbde-d6835808cabf">

`accreditation/21a/pages/05-background-information-chapter/yourBackgroundInformation.js`
<img width="597" alt="Screenshot 2024-06-05 at 11 29 18 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/06389ae1-14e3-46de-8182-9ff6fdb5163c">

`accreditation/21a/pages/06-character-references-chapter/yourCharacterReferences.js`
<img width="597" alt="Screenshot 2024-06-05 at 11 29 35 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/22990386/3d32e59f-3822-4182-aa08-d4a247f0b0a0">

## Acceptance criteria

### Quality Assurance & Testing
